Core/src/org/sleuthkit/autopsy/centralrepository/eventlisteners/CaseEventListener.java

```diff
@@ -154,82 +206,90 @@ private ContentTagTask(CentralRepository db, PropertyChangeEvent evt) {
+
+            if (!isNotableTagName(tagDeletedEvent.getDeletedTagInfo().getName())) {
+                // If the tag that got removed isn't on the list of central repo tags, do nothing
+                return;
+            }
+
+            try {
+                // Get the remaining tags on the content object
                 long contentID = tagDeletedEvent.getDeletedTagInfo().getContentID();
+                Content content = Case.getCurrentCaseThrows().getSleuthkitCase().getContentById(contentID);
+                TagsManager tagsManager = Case.getCurrentCaseThrows().getServices().getTagsManager();
+                List<ContentTag> tags = tagsManager.getContentTagsByContent(content);
 
-                String tagName = tagDeletedEvent.getDeletedTagInfo().getName().getDisplayName();
-                if (!TagsManager.getNotableTagDisplayNames().contains(tagName)) {
-                    // If the tag that got removed isn't on the list of central repo tags, do nothing
+                AbstractFile af = getAbstractFile(content);
+                if (af == null)
                     return;
-                }
 
-                try {
-                    // Get the remaining tags on the content object
-                    Content content = Case.getCurrentCaseThrows().getSleuthkitCase().getContentById(contentID);
-                    TagsManager tagsManager = Case.getCurrentCaseThrows().getServices().getTagsManager();
-                    List<ContentTag> tags = tagsManager.getContentTagsByContent(content);
-
-                    if (tags.stream()
-                            .map(tag -> tag.getName().getDisplayName())
-                            .filter(TagsManager.getNotableTagDisplayNames()::contains)
-                            .collect(Collectors.toList())
-                            .isEmpty()) {
-
-                        // There are no more bad tags on the object
-                        if (content instanceof AbstractFile) {
-                            af = (AbstractFile) content;
-                            knownStatus = TskData.FileKnown.UNKNOWN;
-                            comment = "";
-                        } else {
-                            LOGGER.log(Level.WARNING, "Error updating non-file object");
-                            return;
-                        }
-                    } else {
-                        // There's still at least one bad tag, so leave the known status as is
-                        return;
-                    }
-                } catch (TskCoreException | NoCurrentCaseException ex) {
-                    LOGGER.log(Level.SEVERE, "Failed to find content", ex);
-                    return;
+                if (hasNotableTag(tags)) {
+                    // There's still at least one bad tag, so leave the known status but update the comment to most recent
+                    String comment = getMostRecentNotableComment(tags);
```

Comment:
This logic is not quite correct. Comments in the CR can come from notable tags or directly from the user by way of the add/edit CR comment action. As mentioned in an earlier PR comment, we should clear the comment if and only if the comment on the correlation attribute instance and the tag match. The logic to get a replacement comment from another "notable tag" is a nice innovation - kudos for thinking of it. However, the replacement should only be attempted if we have decided to clear the comment. Otherwise, we could be wiping out a comment from the add/edit action.